{"url":"https://api.github.com/repos/elastic/elasticsearch/issues/26917","repository_url":"https://api.github.com/repos/elastic/elasticsearch","labels_url":"https://api.github.com/repos/elastic/elasticsearch/issues/26917/labels{/name}","comments_url":"https://api.github.com/repos/elastic/elasticsearch/issues/26917/comments","events_url":"https://api.github.com/repos/elastic/elasticsearch/issues/26917/events","html_url":"https://github.com/elastic/elasticsearch/issues/26917","id":263557998,"node_id":"MDU6SXNzdWUyNjM1NTc5OTg=","number":26917,"title":"Unable to delete backup repository","user":{"login":"patmanh","id":1380279,"node_id":"MDQ6VXNlcjEzODAyNzk=","avatar_url":"https://avatars0.githubusercontent.com/u/1380279?v=4","gravatar_id":"","url":"https://api.github.com/users/patmanh","html_url":"https://github.com/patmanh","followers_url":"https://api.github.com/users/patmanh/followers","following_url":"https://api.github.com/users/patmanh/following{/other_user}","gists_url":"https://api.github.com/users/patmanh/gists{/gist_id}","starred_url":"https://api.github.com/users/patmanh/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/patmanh/subscriptions","organizations_url":"https://api.github.com/users/patmanh/orgs","repos_url":"https://api.github.com/users/patmanh/repos","events_url":"https://api.github.com/users/patmanh/events{/privacy}","received_events_url":"https://api.github.com/users/patmanh/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":1,"created_at":"2017-10-06T20:08:25Z","updated_at":"2017-10-10T07:17:25Z","closed_at":"2017-10-10T07:17:24Z","author_association":"NONE","active_lock_reason":null,"body":"<!-- Bug report -->\r\n\r\n**Elasticsearch version** (`bin/elasticsearch --version`):\r\n\r\n`Version: 2.1.1, Build: 40e2c53/2015-12-15T13:05:55Z, JVM: 1.7.0_79`\r\n\r\n**Plugins installed**: []\r\n\r\n```\r\n    - kopf\r\n    - license\r\n    - cloud-aws\r\n    - marvel-agent\r\n```\r\n\r\n**JVM version** (`java -version`):\r\n\r\n```\r\njava version \"1.7.0_79\"\r\nOpenJDK Runtime Environment (IcedTea 2.5.6) (7u79-2.5.6-0ubuntu1.14.04.1)\r\nOpenJDK 64-Bit Server VM (build 24.79-b02, mixed mode)\r\n```\r\n\r\n**OS version** (`uname -a` if on a Unix-like system):\r\n\r\n`Linux ip-10-0-0-230 3.13.0-48-generic #80-Ubuntu SMP Thu Mar 12 11:16:15 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux`\r\n\r\n**Description of the problem including expected versus actual behavior**:\r\n\r\nWe're trying to delete a (very) old s3 backup repository that was set up years ago. There seems to be a 'stuck' restore that is using a snapshot that was in the repo we're trying to delete (the snapshot has since been deleted) . We can see the 'stuck' restore when we hit `http://elasticsearch.hive.co:9200/_cluster/state/customs`\r\n\r\nWe think the stuck repo is preventing us from deleting the repo. We can create new repos and new snapshots OK. Everything else seems to be functioning just fine.\r\n\r\nWe want to delete the bucket that is backing the repo from S3 for cost saving purposes - **is it safe for us to manually delete the repo from S3**? Below are various delete/cancel calls that seem to be failing.\r\n\r\n**Steps to reproduce**:\r\n\r\n```python\r\nes.snapshot.delete(repository='hive-es-backups-repo', snapshot='es-snapshot-2017-07-14-1530')\r\n\r\nConnectionTimeout: ConnectionTimeout caused by - ReadTimeoutError(HTTPConnectionPool(host='ec2-54-86-65-23.compute-1.amazonaws.com', port=9200): Read timed out. (read timeout=10))\r\n```\r\n\r\n```python\r\nresp = requests.delete('http://elasticsearch.hive.co:9200/_snapshot/hive-es-backups-repo/es-snapshot-2017-07-14-1530/')\r\n\r\nprint resp.content\r\n\r\n'{\"error\":{\"root_cause\":[{\"type\":\"snapshot_missing_exception\",\"reason\":\"[hive-es-backups-repo:es-snapshot-2017-07-14-1530] is missing\"}],\"type\":\"snapshot_missing_exception\",\"reason\":\"[hive-es-backups-repo:es-snapshot-2017-07-14-1530] is missing\",\"caused_by\":{\"type\":\"file_not_found_exception\",\"reason\":\"Blob object [snap-es-snapshot-2017-07-14-1530.dat] not found: The specified key does not exist. (Service: Amazon S3; Status Code: 404; Error Code: NoSuchKey; Request ID: F80FD0AA460D476B)\"}},\"status\":404}'\r\n```\r\n\r\n```python\r\nresp = requests.delete('http://elasticsearch.hive.co:9200/_snapshot/hive-es-backups-repo/')\r\n\r\nprint resp.content\r\n\r\n'{\"error\":{\"root_cause\":[{\"type\":\"remote_transport_exception\",\"reason\":\"[MASTER-ip-10-0-0-71][10.0.0.71:9300][cluster:admin/repository/delete]\"}],\"type\":\"illegal_state_exception\",\"reason\":\"trying to modify or unregister repository that is currently used \"},\"status\":500}'\r\n```\r\n\r\nWe've tried to post to the Elasticsearch support forum as well (https://discuss.elastic.co/t/possible-to-manually-delete-s3-backup-repository/101402), with no luck. Any help is appreciated! Thanks.\r\n\r\n","closed_by":{"login":"tlrx","id":642733,"node_id":"MDQ6VXNlcjY0MjczMw==","avatar_url":"https://avatars1.githubusercontent.com/u/642733?v=4","gravatar_id":"","url":"https://api.github.com/users/tlrx","html_url":"https://github.com/tlrx","followers_url":"https://api.github.com/users/tlrx/followers","following_url":"https://api.github.com/users/tlrx/following{/other_user}","gists_url":"https://api.github.com/users/tlrx/gists{/gist_id}","starred_url":"https://api.github.com/users/tlrx/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/tlrx/subscriptions","organizations_url":"https://api.github.com/users/tlrx/orgs","repos_url":"https://api.github.com/users/tlrx/repos","events_url":"https://api.github.com/users/tlrx/events{/privacy}","received_events_url":"https://api.github.com/users/tlrx/received_events","type":"User","site_admin":false},"performed_via_github_app":null}